File: domain/src/main/java/com/rubik/shifttest/domain/domain/usecases/ValidateFirstNameUseCase.java
package com.rubik.shifttest.domain.domain.usecases;


public class ValidateFirstNameUseCase implements Validator {
    private static final String FIRST_NAME_REGEX = "[A-Z][a-z]*";
    @Override
    public boolean execute(CharSequence charSequence) {
                return charSequence.toString().matches(FIRST_NAME_REGEX);
    }
}

File: feature/user/registration/src/main/kotlin/ru/rubik/registration/domain/usecases/ValidateFirstNameUseCase.kt
package ru.rubik.registration.domain.usecases

import javax.inject.Inject

class ValidateFirstNameUseCase @Inject constructor() {
	companion object {

		val FIRST_NAME_REGEX = Regex("[A-Z][a-z]*")
	}

	operator fun invoke(name: String): Boolean = name.matches(FIRST_NAME_REGEX)
}